### PR TITLE
CXP-1067: fix GetAllTestAppsQuery

### PR DESCRIPTION
**Pull request by tseho**
**Description (for Contributor and Core Developer)**

The join was useless and returning inconsistent results.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/GetAllTestAppsQuery.php [hunk 1]
```diff
@@ -31,8 +31,7 @@ public function execute(): GetAllTestAppsResult
     (
         SELECT COUNT(connected_app.id)
         FROM akeneo_connectivity_connected_app connected_app
-        JOIN akeneo_connectivity_connection connection on connection.code = connected_app.connection_code
-        JOIN pim_api_client client on client.marketplace_public_app_id = app.client_id
+        WHERE connected_app.id = app.client_id
```
Comment (masacc) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/GetAllTestAppsQuery.php:34]: no impacts on tests ?

**Review by masacc — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/GetAllTestAppsQuery.php (hunk 1 above)
Comment (tseho) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/GetAllTestAppsQuery.php:34]: I checked and it seems not.
If you want to check the integration test and tell me where it failed to detect the issue, I would really appreciate it :sweat_smile: 
https://github.com/akeneo/pim-community-dev/blob/CXP-1015/src/Akeneo/Connectivity/Connection/back/tests/Integration/Marketplace/GetAllTestAppsQueryIntegration.php

**Comment by RiffVentura** [comment]
Should be done [here](https://github.com/akeneo/pim-community-dev/blob/a472fe8265f6836b76974331913a4d9e9a73a46b/src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/GetTestAppQuery.php#L43) as well

**Review by RiffVentura — APPROVED** [review]

**Comment by tseho** [comment]
> Should be done here as well

good catch, thanks

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/GetAllTestAppsQuery.php (hunk 1 above)
Comment (masacc) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/GetAllTestAppsQuery.php:34]: I take a look at the integration test and it seems logic to work well. :+1:

**Pull request merged**